modules/nf-core/blast/blastn/main.nf

```diff
@@ -21,13 +21,20 @@ process BLAST_BLASTN {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def is_compressed = fasta.name.endsWith(".gz")
+    def fasta_name = fasta.name.replace(".gz", "")
+
     """
+    if [ "${is_compressed}" == "true" ]; then
+        gzip -c -d ${fasta} > ${fasta_name}
+    fi
+
```

Comment:
If we're going to do this, can we pipe it instead? https://edwards.flinders.edu.au/how-to-blast-using-a-compressed-fasta-file/

Might be cleaner to do two different script sections.